analyses/equivalency_cliques.py

```diff
@@ -0,0 +1,104 @@
+"""
+import os
+from typing import List
+
+import pandas as pd
+
+
+CONFIG = {
+    'term_ids': [
+        146910,
+        147010,
+        147070,
+        146910,
+        147070,
+        147010,
+        300015,
+        402500,
+        300151,
+        403000,
+        300162,
+        400011,
+        300357,
+        400023,
+        312095,
+        465000,
+        400020,
+        312865,
+        425000,
+        306250,
+        430000,
+        308385,
+        450000,
+        313470,
+        609517,
+        610067,
+    ],
+    'data_dir_path': os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', 'data')
+}
+
+
+# TODO:
+def run(term_ids: List[int], data_dir_path: str, pk='MIM Number') -> pd.DataFrame:
+    """run analysis
+
+    term_ids: List of OMIM ids.
+    data_dir_path: Path where all the data files are stored.
+    pk: Common primary key used in all data files.
+    """
+    # Collect data
+    table_path__mim_titles = os.path.join(data_dir_path, 'mimTitles.tsv')
+    table_path__mim2gene = os.path.join(data_dir_path, 'mim2gene.tsv')
+    table_path__genemap2 = os.path.join(data_dir_path, 'genemap2.tsv')
+    table_path__allelic_variants = os.path.join(data_dir_path, 'allelicVariants.tsv')
+    table_path__morbidmap = os.path.join(data_dir_path, 'morbidmap.tsv')
+    table_path__phenotypic_series = os.path.join(data_dir_path, 'phenotypicSeries.tsv')
+    table__mim_titles = pd.read_csv(table_path__mim_titles, sep='\t', comment='#')
+    table__mim2gene = pd.read_csv(table_path__mim2gene, sep='\t', comment='#')
+    # todo: utilize other files if/as needed
+    table__genemap2 = pd.read_csv(table_path__genemap2, sep='\t', comment='#')
+    table__allelic_variants = pd.read_csv(table_path__allelic_variants, sep='\t', comment='#')
+    table__morbidmap = pd.read_csv(table_path__morbidmap, sep='\t', comment='#')
+    table__phenotypic_series = pd.read_csv(table_path__phenotypic_series, sep='\t', comment='#')
+
+    # todo: table__mim2gene: entrez gene ID should be int or string without .0's
+    # Formatting
+    pass
+
+    # Initialize dataframe and filter what's needed
+    df = table__mim_titles[table__mim_titles['MIM Number'].isin(term_ids)]
+    table_header__mim_titles = list(df.columns)
+    table_header__mim_titles.remove(pk)
+    df = df[[pk] + table_header__mim_titles]
+
+    # JOIN primary gene data
+    df = pd.merge(df, table__mim2gene, on=pk, how='left')
+    gene_field = 'Approved Gene Symbol (HGNC)'
+    # TODO: move gene_field after MIM Number
+    common_gene_field = 'Terms w common HGNC symbol'
+    # TODO: add col: 'Terms w common HGNC symbol'
+    #  - make it ;-delimited
+    #  - put it after gene_field
+
+    # TODO: decide if I want to include any other data at this time
+
+    # Save and return
+    df.to_csv(os.path.join(data_dir_path, '..', 'equivalency_cliques.csv'), index=False)
```

Comment:
can i see the table?

Reply:
@matentzn Here's what it looks like now:
[equivalency_cliques.csv](https://github.com/monarch-initiative/omim/files/8920812/equivalency_cliques.csv)

@sabrinatoro @nicolevasilevsky: @'ing you guys because Nico may ask one or both of you to look over / work w/ this list.

Note that I did not JOIN all columns in all files, so there is potentially more information (genetic and otherwise) that I can add to this. But I wasn't sure if it was necessary.

The data so far comes from:
1. `mimTitles.txt`
2. `mim2gene.txt`

I have not included but can add if desired:
3. `genemap2.txt`
4. `allelicVariants.txt`
5. `morbidmap.txt`
6. `phenotypicSeries.txt`

Reply:
@matentzn Here's what it looks like now:
[equivalency_cliques.csv](https://github.com/monarch-initiative/omim/files/8920812/equivalency_cliques.csv)

@sabrinatoro @nicolevasilevsky: @'ing you guys because Nico may ask one or both of you to look over / work w/ this list.

Note that I did not JOIN all columns in all files, so there is potentially more information (genetic and otherwise) that I can add to this. But I wasn't sure if it was necessary.

The data so far comes from:
1. `mimTitles.txt`
2. `mim2gene.txt`

I have not included but can add if desired:
3. `genemap2.txt`
4. `allelicVariants.txt`
5. `morbidmap.txt`
6. `phenotypicSeries.txt`

Reply:
Excellent.

Reply:
Add to meeting agenda tomrorrow, so we can tell Nicole what to do with this.

Reply:
If you could write a small paragraph based on the comments in your script that describes:

1. the files you used
2. how you determined the equivalents

We can then stick this into the email to Ada to ask how to best deal with these.

Reply:
Alright. That's semi-described above, but here's a description you can use:

> Joe Flack has done an analysis on OMIM data to ascertain potential problematic class equivalencies. The operating assumption here is that MIM entries of type "gene" that are linked to a common HGNC symbol are potentially problematic. For example, HGNC symbol "IGH" is shared by 3 different MIM #s:
> 1. MIM # 146910: IMMUNOGLOBULIN HEAVY CHAIN DIVERSITY GENE CLUSTER; IGHD@
> 2. MIM # 147070: IMMUNOGLOBULIN HEAVY CHAIN VARIABLE GENE CLUSTER; IGHV@
> 3. MIM # 147010: IMMUNOGLOBULIN HEAVY CHAIN JOINING GENE CLUSTER; IGHJ@
> The files used in this analysis were `mimTitles.txt` and `mim2gene.txt`.

---

Honestly @matentzn @cmungall after writing the above paragraph and looking at the MIM titles, my assumption now is that we have been falsely ascertaining these as equivalent. Looking at this again, I don't think we should be using `owl:equivalentClass` or `skos:exactMatch`, but we should probably use `rdfs:subClassOf` or, probably better, `skos:narrowerThan`. We can look through this CSV that I've made

Reply:
We decided to raise this to Ada.